digiroad2-geo/src/main/scala/fi/liikennevirasto/digiroad2/TrafficSign.scala

```diff
@@ -222,58 +233,136 @@ trait RegulatorySignsType extends TrafficSignType {
 
   override def relevantAdditionalPanel: Seq[AdditionalPanelsType] = supportedAdditionalPanel ++
     Seq(AdditionalPanelWithText, DistanceWhichSignApplies, DistanceFromSignToPointWhichSignApplies, ValidMonFri, ValidSat, ValidMultiplePeriod)
+
+  override def isSpeedLimit: Boolean = {
+    val speedLimitsSign = Seq(5, 6)
+    speedLimitsSign.contains(OTHvalue)
+  }
 }
 
 case object PedestrianCrossingSign extends RegulatorySignsType {
   override val OTHvalue = 7
   override val TRvalue = 511
+  override val OldLawCode = "511"
   override val NewLawCode = "E1"
 }
 
-case object ParkAndRide extends RegulatorySignsType { //TODO check that name
+case object ParkingLot extends RegulatorySignsType {
+  override val OTHvalue = 105
+  override val TRvalue = 521
+  override val OldLawCode = "521"
+  override val NewLawCode = "E2"
+
+  override val supportedAdditionalPanel: Seq[AdditionalPanelsType] = Seq(SignAppliesBothDirections, SignAppliesBothDirectionsVertical,
+    SignAppliesArrowDirections, RegulationBeginsFromSign, RegulationEndsToTheSign,ParkingAgainstFee,	ObligatoryUseOfParkingDisc)
+}
+
+case object ParkingLotAndAccessToTrain extends RegulatorySignsType {
   override val OTHvalue = 137
-  override val TRvalue = 520
-  override val NewLawCode = "E3" //TODO: new sign implements 4 different types
+  override val TRvalue = 520 //5301
```

Comment:
Are these type of comments on the traffic signs related with the new TR Value for the  TR import? If so then maybe it should be better to put that in the comment like //new TRValue 5301 for posterity

Reply:
Updated values and added  better comment